### PR TITLE
Unify virtual machine documented examples

### PR DESCRIPTION
**Pull request by sdake**
Depends-On: https://github.com/istio/istio.io/pull/7248

This PR de-dups the examples and unifies them into one example page.

### CODE REVIEW

**Comment by sdake** [comment]
This work is currently blocked behind the Depends-On: #7248.

**Comment by sdake** [comment]
Work is no longer blocked. Will re-start this work today.

Cheers
-steve

**Comment by sdake** [comment]
cc @tariq1890 you had made some changes in this area. Let's work together on this if you like.

Cheers
-steve

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
-title: Bookinfo with a Virtual Machine
-description: Run the Bookinfo application with a MySQL service running on a virtual
+title: Examples of using virtual machine integration with Istio
```
Comment (tariq1890) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:2]: My understanding was that these tutorials were aimed at adding non-k8s services to a mesh. Maybe we should title it as such?

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+description: Send and receive messages, use Bookinfo application with a MySQL service running on a virtual
```
Comment (tariq1890) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:3]:
```suggestion
description: Send and receive messages, use the Bookinfo application with a MySQL service running on a virtual
```

File: content/en/docs/examples/virtual-machines/bookinfo/index.md (hunk 1 above)
Comment (tariq1890) [reply content/en/docs/examples/virtual-machines/bookinfo/index.md:2]: Adding external services to the mesh, perhaps?

As an Istio user, I am much more likely use that as a search query and stumble upon these docs.

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 3]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+$ echo "10.55.246.247 productpage.default.svc.cluster.local" | sudo tee -a /etc/hosts
+$ curl -v productpage.default.svc.cluster.local:9080
+< HTTP/1.1 200 OK
+< content-type: text/html; charset=utf-8
+< content-length: 1836
+< server: envoy
+... html content ...
+    {{< /text >}}
+
+The `server: envoy` header indicates that the sidecar intercepted the traffic.
```
Comment (tariq1890) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:44]:
```suggestion
The `server: envoy` header indicates that the sidecar has intercepted the traffic.
```

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 4]
```diff
@@ -12,6 +12,151 @@ aliases:
...
 
+## Send requests from VM workloads to Kubernetes services
+
+After setup, the machine can access services running in the Kubernetes cluster
+or on other VMs.
+
+The following example shows accessing a service running in the Kubernetes cluster from a VM using
+`/etc/hosts/`, in this case using a service from the [Bookinfo example](/docs/examples/bookinfo/).
+
+1.  First, on the cluster admin machine get the virtual IP address (`clusterIP`) for the service:
```
Comment (tariq1890) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:23]: First is a tad redundant given we are using a numbered list here.

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 5]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+< HTTP/1.1 200 OK
+< content-type: text/html; charset=utf-8
+< content-length: 1836
+< server: envoy
+... html content ...
+    {{< /text >}}
+
+The `server: envoy` header indicates that the sidecar intercepted the traffic.
+
+## Running services on the added VM
```
Comment (tariq1890) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:46]: Maybe something "starting up and running your external service"?

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 6]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+
+1. Setup an HTTP server on the VM instance to serve HTTP traffic on port 8080:
```
Comment (tariq1890) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:48]: Add a warning note that a GCE account is required here ?

File: content/en/docs/examples/virtual-machines/bookinfo/index.md (hunk 3 above)
Comment (tariq1890) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:44]:
```suggestion
The `server: envoy` header indicates that the sidecar has intercepted the traffic.
```

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 7]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+     `ps`, filtered for `istio`:
+
+    {{< text bash >}}
+    $ ps aux | grep istio
+    root      6955  0.0  0.0  49344  3048 ?        Ss   21:32   0:00 su -s /bin/bash -c INSTANCE_IP=10.150.0.5 POD_NAME=demo-vm-1 POD_NAMESPACE=vm exec /usr/local/bin/pilot-agent proxy > /var/log/istio/istio.log istio-proxy
+    istio-p+  7016  0.0  0.1 215172 12096 ?        Ssl  21:32   0:00 /usr/local/bin/pilot-agent proxy
+    istio-p+  7094  4.0  0.3  69540 24800 ?        Sl   21:32   0:37 /usr/local/bin/envoy -c /etc/istio/proxy/envoy-rev1.json --restart-epoch 1 --drain-time-s 2 --parent-shutdown-time-s 3 --service-cluster istio-proxy --service-node sidecar~10.150.0.5~demo-vm-1.vm-vm.svc.cluster.local
+    {{< /text >}}
+
+-    Check the Envoy access and error logs:
```
Comment (tariq1890) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:154]: We could link the reader to the troubleshooting wiki here

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 8]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+    <h2>Directory listing for /</h2>
+    <hr>
+    <ul>
+    <li><a href=".bashrc">.bashrc</a></li>
+    <li><a href=".ssh/">.ssh/</a></li>
+    ...
+    </body>
+    {{< /text >}}
+
+**Congratulations!** You successfully configured a service running in a pod within the cluster to
```
Comment (tariq1890) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:104]: I prefer the perfect tense here.
```suggestion
**Congratulations!** You have successfully configured a service running in a pod within the cluster to
```

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 9]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+send traffic to a service running on a VM outside of the cluster and tested that
+the configuration worked.
+
+## Cleanup
+
+Run the following commands to remove the expansion VM from the mesh's abstract
```
Comment (tariq1890) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:110]: Not a big fan of "expansion VM" as an outside reader. I still prefer "external service". We could also call the VM that contains the external service, the "expansion VM" earlier on to fill the reader in maybe?

Comment (sdake) [reply content/en/docs/examples/virtual-machines/bookinfo/index.md:110]: I see where the challenge is after reading your review with the word 'integrate with Virtual Machine". How about "Integrate with an external service running in a Virtual machine"

**Comment by sdake** [comment]
Thank you for the reviews. They have helped change my perspective. Additionally, I have done practically zero editings on the content. I hope to get to that this afternoon.

Cheers
-steve

File: content/en/docs/examples/virtual-machines/bookinfo/index.md (hunk 6 above)
Comment (elysdir) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:48]:
```suggestion
1. Set up an HTTP server on the VM instance to serve HTTP traffic on port 8080:
```
(https://developers.google.com/style/word-list#setup)

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 10]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+
+1. Setup an HTTP server on the VM instance to serve HTTP traffic on port 8080:
+
+    {{< text bash >}}
+    $ gcloud compute ssh ${GCE_NAME}
+    $ python -m SimpleHTTPServer 8080
+    {{< /text >}}
+
+1. Determine the VM instance's IP address. For example, find the IP address
+    of the GCE instance with the following commands:
```
Comment (elysdir) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:56]:
```suggestion
1. Determine the VM instance's IP address. For example, to find the IP address
    of the GCE instance, use the following commands:
```
(Fine to leave it as you have it, but I think it's a little clearer with the "to".)

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 11]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+
+    {{< text bash >}}
+    $ export GCE_IP=$(gcloud --format="value(networkInterfaces[0].networkIP)" compute instances describe ${GCE_NAME})
+    $ echo ${GCE_IP}
+    {{< /text >}}
+
+1. Add VM services to the mesh
```
Comment (elysdir) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:63]:
```suggestion
1. Add VM services to the mesh:
```

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 12]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+
+    {{< text bash >}}
+    $ istioctl experimental add-to-mesh external-service vmhttp ${VM_IP} http:8080 -n ${SERVICE_NAMESPACE}
+    {{< /text >}}
+
+    {{< tip >}}
+    Ensure you have added the `istioctl` client to your path, as described in the [download page](/docs/setup/getting-started/#download).
```
Comment (elysdir) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:70]:
```suggestion
    Ensure that you have added the `istioctl` client to your path, as described in the [download page](/docs/setup/getting-started/#download).
```

(https://developers.google.com/style/pronouns#optional-pronouns)

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 13]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+Run the following commands to remove the expansion VM from the mesh's abstract
+model.
+
+{{< text bash >}}
+$ istioctl experimental remove-from-mesh -n ${SERVICE_NAMESPACE} vmhttp
+Kubernetes Service "vmhttp.vm" has been deleted for external service "vmhttp"
+Service Entry "mesh-expansion-vmhttp" has been deleted for external service "vmhttp"
+{{< /text >}}
+
+## Troubleshooting
```
Comment (elysdir) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:119]: Should this troubleshooting  section come after the existing material about Bookinfo?

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 14]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+
+The following are some basic troubleshooting steps for common VM-related issues.
+
+-    When making requests from a VM to the cluster, ensure you don't run the requests as `root` or
+    `istio-proxy` user. By default, Istio excludes both users from interception.
```
Comment (elysdir) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:124]:
```suggestion
-    When making requests from a VM to the cluster, ensure that you don't run the requests as the users `root` or
    `istio-proxy`. By default, Istio excludes both users from interception.
```
You don't have to make exactly this change, but I feel like the phrase "`root` or `istio-proxy` user" is a little awkward/confusing, so I recommend changing it a little, even if you don't make exactly the change I'm suggesting.

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 15]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+
+-    Verify the machine can reach the IP of the all workloads running in the cluster. For example:
```
Comment (elysdir) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:126]:
```suggestion
-    Verify that the machine can reach the IP of all of the workloads running in the cluster. For example:
```
Also, is there more than one IP? If so, then change "IP" to "IPs".

File: content/en/docs/examples/virtual-machines/bookinfo/index.md [hunk 16]
```diff
@@ -12,6 +12,151 @@ aliases:
...
+    ...
+    {{< /text >}}
+
+1. Send a request from the `sleep` service on the pod to the VM's HTTP service:
+
+    {{< text bash >}}
+    $ kubectl exec -it sleep-88ddbcfdd-rm42k -c sleep -- curl vmhttp.${SERVICE_NAMESPACE}.svc.cluster.local:8080
+    {{< /text >}}
+
+    You should see something similar to the output below.
```
Comment (elysdir) [inline content/en/docs/examples/virtual-machines/bookinfo/index.md:89]: We recommend avoiding using the word "should" where possible. (https://developers.google.com/style/word-list#should)

**Comment by frankbu** [comment]
Closing this PR since there have been no updates in over a month. Please fix the conflicts, make updates, and when ready, reopen for further review.

**Pull request closed without merging**